core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -282,13 +286,19 @@ private[spark] class DAGScheduler(
       None
     }
 
-  // Use multi-threaded scheduled executor. The merge finalization task could take some time,
-  // depending on the time to establish connections to mergers, and the time to get MergeStatuses
-  // from all the mergers.
+  // Use multi-threaded scheduled executor. The merge finalization task (per stage) takes up to
+  // PUSH_BASED_SHUFFLE_MERGE_RESULTS_TIMEOUT.
   private val shuffleMergeFinalizeScheduler =
     ThreadUtils.newDaemonThreadPoolScheduledExecutor("shuffle-merge-finalizer",
       shuffleMergeFinalizeNumThreads)
 
+  // The merge finalization task (per stage) will submit a asynchronous thread to send finalize
+  // RPC to the merger location and then get MergeStatus from the merger. This thread won't stop
+  // after PUSH_BASED_SHUFFLE_MERGE_RESULTS_TIMEOUT.
+  private val shuffleSendFinalizeRPCContext =
+    ExecutionContext.fromExecutor(ThreadUtils.newDaemonFixedThreadPool(
+      shuffleSendFinalizeRPCThreads, "send-shuffle-merge-finalize-rpc"))
```

Comment:
Do we really need more than 1 thread for this?

Reply:
According to @mridulm comments, we will submit a task for each finalize RPC to one merger node. If one task tries to create a connection to a merger node, the other tasks won't be blocked, so we can get most of the merged statuses.
If there are `shuffleSendFinalizeRPCThreads` tasks trying to create a connection to a merger node, the corresponding stages will be blocked for `SPARK_NETWORK_IO_CONNECTIONCREATIONTIMEOUT`

Reply:
Ok, I missed the comments. Why did you chose 6 as the default for this config?

Reply:
Because I think the number of send RPC tasks and total elapsed time in `shuffleSendFinalizeRPCContext` should bigger than tasks in `shuffleMergeFinalizeScheduler`.

Reply:
Because I think the number of send RPC tasks and total elapsed time in `shuffleSendFinalizeRPCContext` should bigger than tasks in `shuffleMergeFinalizeScheduler`.